include/soci/use-type.h

```diff
@@ -72,6 +72,21 @@ class SOCI_DECL standard_use_type : public use_type_base
         //convert_to_base();
     }
 
+    // Const indicator constructor
+    standard_use_type(void* data, exchange_type type, const indicator& ind,
+        bool readOnly, std::string const& name = std::string())
+        : data_(data)
+        , type_(type)
+        // Const cast will allow the code to compile and run but any modification to ind_ in readonly mode will result in undefined behaviour
```

Comment:
This is not very reassuring... Especially because whether `ind_` is modified or not is controlled by `readOnly` so there seems to be a possibility of constructing this object with const indicator but `readOnly == false` which would result in disaster.

IMHO we shouldn't have `readOnly` parameter in this ctor overload at all, it should always be true...